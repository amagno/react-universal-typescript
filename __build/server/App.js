"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const MakeDocument_1 = require("./Lib/MakeDocument");
const handleRender = (req, res) => {
    res.send(MakeDocument_1.default('Hello World'));
};
const configureRender = (app, path = '*') => {
    return app.get(path, handleRender);
};
class App {
    constructor(app, options) {
        this.app = configureRender(app);
    }
    express() {
        return this.app;
    }
}
exports.default = App;
