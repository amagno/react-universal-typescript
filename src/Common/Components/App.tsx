import { Route, Switch, RouteProps } from 'react-router';
import * as React from 'react';

interface AppProps {
  routes: RouteProps[];
}
export default class App extends React.Component<AppProps> {
  render () {
    return (
      <div>
        <h1>Hello World!</h1>
        <Switch>
          {this.props.routes.map((route, index) => (
            <Route key={index} {...route} />
          ))}
        </Switch>
      </div>
    );
  }
}