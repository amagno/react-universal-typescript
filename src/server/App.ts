import * as express from 'express';
import MakeDocument from './Lib/MakeDocument';
const handleRender = (req: express.Request, res: express.Response) => {
  res.send(MakeDocument('Hello World'));
};
const configureRender = (app: express.Application, path: string = '*'): express.Application => {
  return app.get(path, handleRender);
};
class App {
  private app: express.Application;
  constructor(app: express.Application, options?: object) {
    this.app = configureRender(app);
  }
  express() {
    return this.app;
  }
}
export default App;