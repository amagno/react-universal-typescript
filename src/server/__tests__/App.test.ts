import 'jest';
import * as supertest from 'supertest';
import App from '../App';
import * as express from 'express';
import { Routes, resolve } from 'universal-router';
const UniversalRouter = require('universal-router');



describe('test app is running', () => {
  it('it run app with routes', async () => {
    const routes: express.Application = express();
    routes.get('/', (req, res) => res.send('hello world!'));
    const app = new App(routes);
    const req = await supertest(app.express()).get('/');
    expect(req.text).toMatchSnapshot();
  });
  it('resolve router', async () => {
    const universalRoutes: Routes<{}, string> = [
      {
        path: '/',
        action: () => 'hello world!'
      }
    ];
    
    const router = new UniversalRouter(universalRoutes);
    const data = await router.resolve({ pathname: '/'});
    console.log(data);
    
    expect(data).toEqual('hello world!');
  });
});