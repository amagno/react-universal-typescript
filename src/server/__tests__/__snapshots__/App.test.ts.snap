// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test app is running it run app with routes 1`] = `"hello world!"`;
